ckcp/docker/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#use golang as base image and compile kcp
+FROM golang:1.17 as src
+
+LABEL usage="using multi-stage build to compile kcp using go"
+
+WORKDIR /go/src/app
+RUN git clone https://github.com/kcp-dev/kcp.git && cd ./kcp && git checkout ae4e07fcd4399264dccab47509ef17f2851050ee
+
+WORKDIR /go/src/app/kcp
+RUN go build -ldflags "-X k8s.io/component-base/version.gitVersion=v1.22.2 -X k8s.io/component-base/version.gitCommit=5e58841cce77d4bc13713ad2b91fa0d961e69192" -o bin/kcp ./cmd/kcp
+
+FROM fedora:latest
```

Comment:
You can use https://hub.docker.com/r/redhat/ubi8/tags images. It will be smaller.

Reply:
I tried this before. After all the cleaning up, here is the docker images sizes. ubi is slightly larger than fedora. If you think ubi is better in the long run, let me know and I will change it.

```
[bnr@bnr docker]$ docker images
REPOSITORY             TAG       IMAGE ID       CREATED             SIZE
quay.io/bnr/ubi-ckcp   latest    98b9e7acff1c   3 seconds ago       317MB
quay.io/bnr/ckcp       latest    c9fb03ad1a3f   About an hour ago   255MB
```